### ISSUE TITLE
Headers are case sensitive in base ApiGateway class

### ISSUE THREAD

**Issue by spencerkittleson** [LambdaSharp/LambdaSharpTool#129]
According to https://tools.ietf.org/html/rfc7230#section-3.2, headers should be case-insensitive.  The headers directory is case-sensitive: `ALambdaApiGatewayFunction.APIGatewayProxyRequest.Headers`.

Suggested change: `var headers = new Dictionary<string, string>(
        StringComparer.InvariantCultureIgnoreCase);`

**Comment by bjorg** [comment]
Great find! Thx!

**Comment by bjorg** [comment]
This appears to be how `APIGatewayProxyRequest` is deserialized. I can add this line to force it to be case-insensitive, but it feels hacky:
```csharp
request.Headers = new Dictionary<string, string>(request.Headers, StringComparer.InvariantCultureIgnoreCase);
```

Thoughts?

**Comment by spencerkittleson** [comment]
That's the workaround i used and it's working perfectly.

**Comment by bjorg** [comment]
Fixed in 0.7.0.8.
